core-android/src/main/java/com/uber/sdk/android/core/auth/LoginManager.java

```diff
@@ -363,4 +410,54 @@ private void handleResultOk(@Nullable Intent data) {
 
         }
     }
+
+    boolean isInLegacyRedirectMode(@NonNull Activity activity) {
+        String generatedRedirectUri = activity.getPackageName().concat(".uberauth://redirect");
+        String setRedirectUri = sessionConfiguration.getRedirectUri();
+
+        if (redirectForAuthorizationCode) {
+            String message = "The Uber Authentication Flow for the Authorization Code Flow has "
+                    + "been upgraded in 0.8.0 and a redirect URI must now be supplied to the application. "
+                    + "You are seeing this error because the use of deprecated method "
+                    + "LoginManager.setRedirectForAuthorizationCode() indicates your flow may not "
+                    + "support the recent changes. See https://github"
+                    + ".com/uber/rides-android-sdk#authentication-migration-version"
+                    + "-08-and-above for resolution steps"
+                    + "to insure your setup is correct and then migrate to the non-deprecate "
+                    + "method LoginManager.setAuthCodeFlowEnabled()";
+
+            logOrError(activity, new IllegalStateException(message));
+            return true;
+        }
+
+        if (sessionConfiguration.getRedirectUri() == null) {
```

Comment:
It's technically possible to use auth without a redirect URI when using SSO, but because we can't guarantee the clients all have the Uber app installed, it's better to play it safer with overall auth and fail faster up front. We've added the check in login, as well as this explicit check, in case a developer was using one of the explicit login methods directly.